### PR TITLE
Fix getting partitions in Proxy meta cache

### PR DESCRIPTION
**Pull request by bigsheeper**
issue: https://github.com/milvus-io/milvus/issues/19596

/kind bug

### CODE REVIEW

File: internal/proxy/meta_cache.go [hunk 1]
```diff
@@ -350,19 +350,12 @@ func (m *MetaCache) GetPartitionID(ctx context.Context, collectionName string, p
 }
 
 func (m *MetaCache) GetPartitions(ctx context.Context, collectionName string) (map[string]typeutil.UniqueID, error) {
-	_, err := m.GetCollectionID(ctx, collectionName)
+	collInfo, err := m.GetCollectionInfo(ctx, collectionName)
 	if err != nil {
 		return nil, err
 	}
 
 	m.mu.RLock()
```
Comment (longjiquan) [inline internal/proxy/meta_cache.go:358]: It seems that `lock` is not necessary here.

File: internal/proxy/meta_cache.go [hunk 2]
```diff
@@ -403,21 +396,13 @@ func (m *MetaCache) GetPartitions(ctx context.Context, collectionName string) (m
 }
 
 func (m *MetaCache) GetPartitionInfo(ctx context.Context, collectionName string, partitionName string) (*partitionInfo, error) {
-	_, err := m.GetCollectionID(ctx, collectionName)
+	collInfo, err := m.GetCollectionInfo(ctx, collectionName)
 	if err != nil {
 		return nil, err
 	}
 
 	m.mu.RLock()
```
Comment (longjiquan) [inline internal/proxy/meta_cache.go:404]: Ditto.

**Review by longjiquan — CHANGES_REQUESTED** [review]

File: internal/proxy/meta_cache.go (hunk 1 above)
Comment (bigsheeper) [reply internal/proxy/meta_cache.go]: updated

**Pull request merged**